backend/python/app/services/implementations/story_service.py

```diff
@@ -707,32 +707,26 @@ def finish_grading_story_translation(
                 .all()
             )
 
-            translate_passed = test_result["translate"]
-            review_passed = test_result["review"]
+            translate_level = test_result["translate"]
+            review_level = test_result["review"]
 
-            if translate_passed or review_passed:
+            if translate_level or review_level:
                 language = story_translation_test.language
-                level = (
-                    db.session.query(Story)
-                    .filter(Story.id == story_translation_test.story_id)
-                    .first()
-                    .level
-                )
 
                 user = (
                     db.session.query(User)
                     .filter(User.id == story_translation_test.translator_id)
                     .first()
                 )
 
-                if translate_passed:
+                if translate_level:
                     appr_lang = user.approved_languages_translation
-                    appr_lang[language] = level
+                    appr_lang[language] = translate_level
                     user.approved_languages_translation = appr_lang
 
-                if review_passed:
+                if review_level:
```

Comment:
getting the error `'NoneType' object does not support item assignment` using `"{\"translate\": 4, \"review\": 3 }"` for the `storyTranslationTestId: 14`. 

I think the edgecase missing is if an appr_language is `null`.  I.e, carl sagan has `null` for appr_lang_reviewer, so trying to set `FRENCH` on it throws an error